pkg/controller/session/session_controller_test.go

```diff
@@ -232,6 +245,114 @@ var _ = Describe("Basic session manipulation", func() {
 				Expect(modified.Status.Refs).To(HaveLen(0))
 			})
 		})
+		Context("updated reference", func() {
+			BeforeEach(func() {
+				objects = []runtime.Object{
+					&v1alpha1.Session{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:       "test-session",
+							Namespace:  "test",
+							Finalizers: []string{session.Finalizer},
+						},
+						Spec: v1alpha1.SessionSpec{
+							Refs: []v1alpha1.Ref{
+								{
+									Name:     "details",
+									Strategy: "telepresence",
+								},
+								{
+									Name:     "ratings",
+									Strategy: "prepared-image",
+									Args: map[string]string{
+										"image": "x",
+									},
+								},
+								{
+									Name:     "locations",
+									Strategy: "prepared-image",
+									Args: map[string]string{
+										"image": "y",
+									},
+								}},
+						},
+						Status: v1alpha1.SessionStatus{
+							Refs: []*v1alpha1.RefStatus{
+								{
+									Ref: v1alpha1.Ref{
+										Name:     "details",
+										Strategy: "prepared-image",
+										Args: map[string]string{
+											"image": "x",
+										}},
+									Resources: []*v1alpha1.RefResource{{Kind: &kind, Name: &name, Action: &action}},
+								},
+								{
+									Ref: v1alpha1.Ref{
+										Name:     "ratings",
+										Strategy: "telepresence",
+									},
+									Resources: []*v1alpha1.RefResource{{Kind: &kind, Name: &name, Action: &action}},
+								},
+								{
+									Ref: v1alpha1.Ref{
+										Name:     "locations",
+										Strategy: "prepared-image",
+										Args: map[string]string{
+											"image": "x",
+										}},
+									Resources: []*v1alpha1.RefResource{{Kind: &kind, Name: &name, Action: &action}},
+								},
+							},
+						},
+					},
+				}
+			})
+			It("updated strategy", func() {
```

Comment:
I was wondering if you could give a bit more meaningful description for those tests so that it's understood from the description itself what it the subject of the test. In particular, I don't see how the update is happening in the test code before jumping to the implementation of `basicTestRevertor` (maybe this should be a more action-focused name such as `removeStatus(from, resource)`?).

Reply:
https://github.com/Maistra/istio-workspace/pull/279/commits/db555b712ae70fd1355eb61f05c691896a96054e

Reply:
https://github.com/Maistra/istio-workspace/pull/279/commits/f26a3f0b8319685d6aa2f8d99f9ccf6db211c3ed